bin/server.js

```diff
@@ -1,21 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
 var config = require('../config')
-var createServer = require('fxa-auth-db-server')
-var error = require('../error')
+var dbServer = require('fxa-auth-db-server')
+var error = dbServer.errors
 var log = require('../log')(config.logLevel, 'db-api')
 var DB = require('../db/mysql')(log, error)
-var version = require('../package.json').version
 
 function shutdown() {
   process.nextTick(process.exit)
 }
 
-// defer to allow ass code coverage results to complete processing
-if (process.env.ASS_CODE_COVERAGE) {
-  process.on('SIGINT', shutdown)
-}
+if (require.main === module) {
+  // defer to allow ass code coverage results to complete processing
+  if (process.env.ASS_CODE_COVERAGE) {
+    process.on('SIGINT', shutdown)
+  }
 
-DB.connect(config).done(
-  function (db) {
-    var server = createServer(version, db, log, config.port, config.host)
+  DB.connect(config).done(
+    function (db) {
+      var server = dbServer.createServer(db)
+      server.listen(config.port, config.hostname)
+      server.on('failure', function (d) { console.error(d.err.code, d.url)})
+      server.on('success', function (d) { console.log(d.method, d.url)})
+    }
+  )
+}
+else {
+  module.exports = function () {
+    return DB.connect(config)
```

Comment:
One tiny nit about the structure of this code: you have the connect-then-createServer logic duplicated into both branches of this if/else.  It may be cleaner to unconditionally export the server-creating logic in module.export, and then have the "if running as main" branch just call it as necessary.